core/src/main/java/hudson/triggers/Trigger.java

```diff
@@ -275,7 +275,14 @@ public void run(AbstractProject p) {
                         if (t.tabs.check(cal)) {
                             LOGGER.log(Level.CONFIG, "cron triggered {0}", p);
                             try {
+                                long begin_time = System.currentTimeMillis();
                                 t.run();
+                                long end_time = System.currentTimeMillis();
+                                if ((end_time - begin_time) > CRON_THRESHOLD) {
+                                    LOGGER.log(Level.WARNING, "cron trigger " + t.getClass().getName() + ".run() triggered by {0} spent too much time ({1}) in its execution, " +
+                                            "other timers can be affected",
+                                            new Object[] {p, Util.getTimeSpanString(end_time - begin_time)});
```

Comment:
We should either use plain old string concatenation or parameter replacement. Using both give us disadvantages from both worlds.